library/src/hooks/street-view-panorama.ts

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable complexity */
+import {useMemo, useState} from 'react';
+
+import {useGoogleMap} from './map-instance';
+
+interface StreetViewPanoramaProps {
+  divElement?: HTMLElement | null;
+  position?: google.maps.LatLng | google.maps.LatLngLiteral;
+  heading?: number;
+  pitch?: number;
+  zoom?: number;
+}
+/**
+ * Hook to get Street View Panorama
+ */
+export const useStreetViewPanorama = (
+  props: StreetViewPanoramaProps
+): google.maps.StreetViewPanorama | null => {
+  const {divElement, position, heading, pitch, zoom} = props;
+
+  // Get map instance
+  const {map} = useGoogleMap();
+
+  const [panorama, setPanorama] =
+    useState<google.maps.StreetViewPanorama | null>(null);
```

Comment:
What is this state for? There's already the `streetViewPanorama` in line 28.

Reply:
You hate states, no? :D

It is for line 36 an 57, but I can also return them directly if preferred

Reply:
> but I can also return them directly if preferred 

Yes please.